### ISSUE TITLE
the time of opencv_catkin build reaches a day ? so long long time?

### ISSUE THREAD

**Issue by fangxu622** [ethz-asl/maplab#87]

**Comment by schneith** [comment]
The opencv_catkin package downloads the source code before building. Sometimes this downloads fails and let's the build hang. In that case, it is best to clean your workspace with 'catkin clean' and restart the build.

**Comment by fangxu622** [comment]
@schneith   I downloaded opencv3 sourcecode before building.This situation still occurs

**Comment by ZacharyTaylor** [comment]
How much ram does your computer have? If it is running out (sometimes an issue with < 4gb) this could cause the issue. You could also try building with -j 1 to ensure only one package builds at once.

**Comment by fangxu622** [comment]
RAM  = 8gb

**Comment by fangxu622** [comment]
@ZacharyTaylor  ram 8gb

**Comment by fangxu622** [comment]
@dymczykm  yes, catkin clean , then input "yes" or  "no", We've tried.it always stucked in opencv3_catkin build phase， Sometimes it lasts 4 days in opencv3_catkin build phase .
![image](https://user-images.githubusercontent.com/16681534/44140928-37b24164-a0ae-11e8-8a58-a2bc44e2f1f6.png)

**Comment by fangxu622** [comment]
![image](https://user-images.githubusercontent.com/16681534/44141056-93644340-a0ae-11e8-8474-2558d84ddb20.png)

**Comment by mfehr** [comment]
Hi @fangxu622 

Please clean and rebuild the workspace again, but this time use the following flags:
```bash
catkin build opencv3_catkin -v --interleave-output
```
This will help you to figure out exactly where the build gets stuck. As @schneith  already pointed out, opencv3_catkin can cause trouble if you have a weak internet connection, because it takes a while to download or there might be a timeout. But if the build takes more than an hour on a modern machine it is most likely stuck. So no need to wait for 4 days :)

**Comment by smauq** [comment]
Another possibility from the [FAQ](https://github.com/ethz-asl/maplab/wiki/FAQ) on "Why is my opencv3_catkin build freezing?". As ridiculous at it sounds, apparently this can be an issue:

> A known cause for this is an installed MATLAB using a network licence. The build can stall on getting the licence authenticated; so make sure that the licence server can be accessed, connect to the VPN if required."

**Comment by fangxu622** [comment]
@mfehr  thank you ,now ,this problem occurs, no "opencv_src-configure" files.   the opencv source code is downloaded before buiding,  Is there supposed to be  "opencv_src-configure" file？
![image](https://user-images.githubusercontent.com/16681534/44151085-06a11124-a0d4-11e8-98a1-efd301ea9e30.png)



![image](https://user-images.githubusercontent.com/16681534/44151429-275b7b38-a0d5-11e8-9f85-676b7ec28f39.png)

**Comment by fangxu622** [comment]
![image](https://user-images.githubusercontent.com/16681534/44151548-7ce2fd10-a0d5-11e8-83f1-018b535e3d7c.png)

**Comment by fangxu622** [comment]
@smauq  it 's network problems ,but something else wrong after fixing network problems

**Comment by fangxu622** [comment]
ok, fixed it,thanks all @schneith @ZacharyTaylor @dymczykm @mfehr ,just retry

**Comment by dbHub20092758** [comment]
the same problem , how to fix it. i download opencv package before build @fangxu622
